front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PluginController.java

```diff
@@ -0,0 +1,64 @@
+package com.netflix.spinnaker.front50.controllers;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+import com.netflix.spinnaker.front50.model.plugin.PluginRepository;
+import com.netflix.spinnaker.front50.model.plugin.PluginRepository.GetCriteria;
+import com.netflix.spinnaker.front50.model.plugin.PluginRepository.ListCriteria;
+import com.netflix.spinnaker.front50.model.plugin.PluginRepository.UpsertData;
+import com.netflix.spinnaker.front50.proto.Plugin;
+import com.netflix.spinnaker.front50.proto.Plugins;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.http.converter.HttpMessageConversionException;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(value = "plugins")
+@ConditionalOnExpression("${spinnaker.plugins.enabled:false}")
+public class PluginController {
```

Comment:
Essentially, if the content type is `application/x-protobuf` this controller will produce and consume protobufs.